segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -493,6 +504,13 @@ public boolean isOffline() {
             // the Merge right after the Seal.
             result = CompletableFuture.allOf(result,
                     this.durableLog.add(new MergeSegmentOperation(targetSegmentId, sourceSegmentId), timer.getRemaining()));
+            //if segment is a transaction, add its length onto metrics of target segment
```

Comment:
It is a good idea to place this kind of metrics logic in upper layers whenever possible. For this reason, it would be nice to explore if it is possible to report these metrics in `PravegaRequestProcessor.mergeSegments` (there is a private method `recordStatForTransaction` in which you can add your reporting logic).

Reply:
fixed